src/transformers/testing_utils.py

```diff
@@ -2993,6 +3008,9 @@ def _device_agnostic_dispatch(device: str, dispatch_table: dict[str, Callable],
     BACKEND_EMPTY_CACHE["xpu"] = torch.xpu.empty_cache
     BACKEND_MANUAL_SEED["xpu"] = torch.xpu.manual_seed
     BACKEND_DEVICE_COUNT["xpu"] = torch.xpu.device_count
+    BACKEND_RESET_MAX_MEMORY_ALLOCATED["xpu"] = torch.xpu.reset_peak_memory_stats
```

Comment:
xpu map `reset_peak_memory_stats` to `RESET_MAX_MEMORY_ALLOCATED`, since they have same functionality. [CUDA](https://github.com/pytorch/pytorch/blob/main/torch/cuda/memory.py#L471)'s impl is same. The difference is CUDA keeps the `reset_max_memory_allocated` API for backward-compatibility, XPU doesn't expose this API.